### PR TITLE
[DOCS-5974] CI Visibility Add code coverage guide 

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What does this PR do?
Add a guide to use code coverage in CI Visibility. 

### Motivation
Let users know about this feature and make it easier for them to use it. 

### Preview

https://docs-staging.datadoghq.com/juan-fernandez/code-coverage-guide/continuous_integration/guides/code_coverage/?tab=javascripttypescript

https://docs-staging.datadoghq.com/juan-fernandez/code-coverage-guide/continuous_integration/guides/code_coverage/?tab=net

https://docs-staging.datadoghq.com/juan-fernandez/code-coverage-guide/continuous_integration/guides/code_coverage/?tab=junitreportuploads

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/continuous_integration/guides/code_coverage.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+### Compatibility
+
+* `dd-trace>=3.20.0`.
+* `jest>=24.8.0`, only when run with `jest-circus`.
+* `mocha>=5.2.0`.
+* `cucumber-js>=7.0.0`.
+
+When tests are instrumented with [Istanbul][1], the Datadog Tracer reports code coverage under the `test.code_coverage.lines_pct` tag for your test sessions automatically.
+
+To instrument tests with Istanbul, the easiest way is to use [`nyc`][2]:
```
Comment (alai97) [inline content/en/continuous_integration/guides/code_coverage.md:32]:
```suggestion
To instrument tests with Istanbul, you can use [`nyc`][2]:
```

File: content/en/continuous_integration/guides/code_coverage.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+```json
+{
+  "scripts": {
+    "test": "mocha",
+    "coverage": "nyc npm run test"
+  }
+}
+```
+
+Then use:
```
Comment (alai97) [inline content/en/continuous_integration/guides/code_coverage.md:43]:
```suggestion
Then, use:
```

File: content/en/continuous_integration/guides/code_coverage.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+```
+NODE_OPTIONS="-r dd-trace/ci/init" DD_ENV=ci DD_SERVICE=my-javascript-service npm run coverage
+```
+
+Jest includes istanbul by default, so you can simply do:
```
Comment (alai97) [inline content/en/continuous_integration/guides/code_coverage.md:49]:
```suggestion
Jest includes Istanbul by default, so you can do:
```

File: content/en/continuous_integration/guides/code_coverage.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+```json
+{
+  "scripts": {
+    "test:coverage": "jest --coverage"
+  }
+}
+```
+
+```
```
Comment (alai97) [inline content/en/continuous_integration/guides/code_coverage.md:59]: Is there any instruction you'd like to provide here?

File: content/en/continuous_integration/guides/code_coverage.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+<div class="alert alert-warning">CI Visibility is not available in the selected site ({{< region-param key="dd_site_name" >}}) at this time.</div>
+{{< /site-region >}}
+
+
+## Report code coverage
+
+Before you can follow this guide, make sure that [Test Visibility][1] is already set up for your language.
+
+{{< tabs >}}
+{{% tab "Javascript/Typescript" %}}
```
Comment (alai97) [inline content/en/continuous_integration/guides/code_coverage.md:21]:
```suggestion
{{% tab "JavaScript/TypeScript" %}}
```

File: content/en/continuous_integration/guides/code_coverage.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+---
+
+{{< site-region region="gov" >}}
+<div class="alert alert-warning">CI Visibility is not available in the selected site ({{< region-param key="dd_site_name" >}}) at this time.</div>
+{{< /site-region >}}
+
+
+## Report code coverage
+
+Before you can follow this guide, make sure that [Test Visibility][1] is already set up for your language.
```
Comment (alai97) [inline content/en/continuous_integration/guides/code_coverage.md:18]:
```suggestion
Ensure that [Test Visibility][1] is already set up for your language.
```

File: content/en/continuous_integration/guides/code_coverage.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+[1]: https://istanbul.js.org/
+[2]: https://github.com/istanbuljs/nyc
+{{% /tab %}}
+
+{{% tab ".NET" %}}
+
+### Compatibility
+* `dd-trace>=2.31.0`.
+
+When code coverage is available, the Datadog Tracer (v2.31.0+) reports it under the `test.code_coverage.lines_pct` tag for your test sessions.
```
Comment (alai97) [inline content/en/continuous_integration/guides/code_coverage.md:74]:
```suggestion
When code coverage is available, the Datadog Tracer (v2.31.0 or later) reports it under the `test.code_coverage.lines_pct` tag for your test sessions.
```

File: content/en/continuous_integration/guides/code_coverage.md [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+If you are using [Coverlet][1] to compute your code coverage, indicate the path to the report file in the `DD_CIVISIBILITY_EXTERNAL_CODE_COVERAGE_PATH` environment variable when running `dd-trace`. The report file must be in the OpenCover or Cobertura formats. Alternatively, you can enable the Datadog Tracer’s built-in code coverage calculation with the env var `DD_CIVISIBILITY_CODE_COVERAGE_ENABLED=true`.
+
+
+[1]: https://github.com/coverlet-coverage/coverlet
```
Comment (alai97) [inline content/en/continuous_integration/guides/code_coverage.md:79]:
```suggestion
If you are using [Coverlet][1] to compute your code coverage, indicate the path to the report file in the `DD_CIVISIBILITY_EXTERNAL_CODE_COVERAGE_PATH` environment variable when running `dd-trace`. The report file must be in the OpenCover or Cobertura formats. Alternatively, you can enable the Datadog Tracer’s built-in code coverage calculation with the `DD_CIVISIBILITY_CODE_COVERAGE_ENABLED=true` env variable.


[101]: https://github.com/coverlet-coverage/coverlet
```

File: content/en/continuous_integration/guides/code_coverage.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+{{% /tab %}}
+{{< /tabs >}}
+
+### Graph code coverage
+
+The reported code coverage can be seen in the **Coverage** tab in the test session detail:
```
Comment (alai97) [inline content/en/continuous_integration/guides/code_coverage.md:85]:
```suggestion
Reported code coverage appears on the **Coverage** tab in a test session details page:
```

File: content/en/continuous_integration/guides/code_coverage.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+
+{{< img src="/continuous_integration/code_coverage_tab.png" text="Test sessions code coverage tab" style="width:100%" >}}
+
+
+### Export your graph
+
+It is possible to export your graph to a [dashboard][2] or a [notebook][3] and even create a [monitor][4] based on it. Click **Export** on the right side:
```
Comment (alai97) [inline content/en/continuous_integration/guides/code_coverage.md:92]:
```suggestion
You can export your graph to a [dashboard][2] or a [notebook][3] and create a [monitor][4] based on it. 

Click **Export** on the right side:
```

File: content/en/continuous_integration/tests/dotnet.md [hunk 11]
```diff
@@ -166,20 +166,22 @@ if (scope != null) {
...
 
-If you are using [Coverlet][14] to compute your code coverage, indicate the path to the report file in the `DD_CIVISIBILITY_EXTERNAL_CODE_COVERAGE_PATH` environment variable when running `dd-trace`. The report file must be in the OpenCover or Cobertura formats. Alternatively, you can enable the Datadog Tracer's built-in code coverage calculation with the env var `DD_CIVISIBILITY_CODE_COVERAGE_ENABLED=true`.
+If you are using [Coverlet][10] to compute your code coverage, indicate the path to the report file in the `DD_CIVISIBILITY_EXTERNAL_CODE_COVERAGE_PATH` environment variable when running `dd-trace`. The report file must be in the OpenCover or Cobertura formats. Alternatively, you can enable the Datadog Tracer's built-in code coverage calculation with the env var `DD_CIVISIBILITY_CODE_COVERAGE_ENABLED=true`.
 
 **Note**: When using Intelligent Test Runner, the tracer's built-in code coverage is enabled by default.
 
 You can see the evolution of the test coverage in the **Coverage** tab of a test session.
 
+Read more about code coverage in Datadog in [code coverage in Datadog guide][11].
```
Comment (alai97) [inline content/en/continuous_integration/tests/dotnet.md:183]:
```suggestion
Read more about code coverage in Datadog in the [Code Coverage guide][11].
```

File: content/en/continuous_integration/tests/dotnet.md [hunk 12]
```diff
@@ -166,20 +166,22 @@ if (scope != null) {
...
 
-Read more about custom metrics in the [Add Custom Metrics Guide][15].
+Read more about custom metrics in the [Add Custom Metrics Guide][9].
 
 ### Reporting code coverage
 
 When code coverage is available, the Datadog Tracer (v2.31.0+) reports it under the `test.code_coverage.lines_pct` tag for your test sessions.
 
-If you are using [Coverlet][14] to compute your code coverage, indicate the path to the report file in the `DD_CIVISIBILITY_EXTERNAL_CODE_COVERAGE_PATH` environment variable when running `dd-trace`. The report file must be in the OpenCover or Cobertura formats. Alternatively, you can enable the Datadog Tracer's built-in code coverage calculation with the env var `DD_CIVISIBILITY_CODE_COVERAGE_ENABLED=true`.
+If you are using [Coverlet][10] to compute your code coverage, indicate the path to the report file in the `DD_CIVISIBILITY_EXTERNAL_CODE_COVERAGE_PATH` environment variable when running `dd-trace`. The report file must be in the OpenCover or Cobertura formats. Alternatively, you can enable the Datadog Tracer's built-in code coverage calculation with the env var `DD_CIVISIBILITY_CODE_COVERAGE_ENABLED=true`.
```
Comment (alai97) [inline content/en/continuous_integration/tests/dotnet.md:177]:
```suggestion
If you are using [Coverlet][10] to compute your code coverage, indicate the path to the report file in the `DD_CIVISIBILITY_EXTERNAL_CODE_COVERAGE_PATH` environment variable when running `dd-trace`. The report file must be in the OpenCover or Cobertura formats. Alternatively, you can enable the Datadog Tracer's built-in code coverage calculation with the `DD_CIVISIBILITY_CODE_COVERAGE_ENABLED=true` environment variable.
```

File: content/en/continuous_integration/tests/dotnet.md [hunk 13]
```diff
@@ -65,11 +65,11 @@ Install or update the `dd-trace` command using one of the following ways:
...
 - By downloading the appropriate version:
-    * Win-x64: [https://dtdg.co/dd-trace-dotnet-win-x64][2]
-    * Linux-x64: [https://dtdg.co/dd-trace-dotnet-linux-x64][3]
-    * Linux-musl-x64 (Alpine): [https://dtdg.co/dd-trace-dotnet-linux-musl-x64][4]
+    * Win-x64: [https://dtdg.co/dd-trace-dotnet-win-x64][3]
+    * Linux-x64: [https://dtdg.co/dd-trace-dotnet-linux-x64][4]
+    * Linux-musl-x64 (Alpine): [https://dtdg.co/dd-trace-dotnet-linux-musl-x64][5]
 
-- Or by downloading [from the github release page][5].
+- Or by downloading [from the github release page][6].
```
Comment (alai97) [inline content/en/continuous_integration/tests/dotnet.md:72]:
```suggestion
- Or by downloading [from the GitHub release page][6].
```

File: content/en/continuous_integration/guides/code_coverage.md [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+
+{{< img src="/continuous_integration/code_coverage_export_to.png" text="Export code coverage" style="width:100%" >}}
+
+
+### Add a monitor
+
+Get alerted whenever code coverage for your service drops below a certain threshold:
```
Comment (alai97) [inline content/en/continuous_integration/guides/code_coverage.md:99]:
```suggestion
Get alerted whenever code coverage for your service drops below a certain threshold using a CI Test Monitor:
```

File: content/en/continuous_integration/guides/code_coverage.md [hunk 15]
```diff
@@ -0,0 +1,111 @@
...
+
+{{< img src="/continuous_integration/code_coverage_monitor.png" text="Test sessions code coverage tab" style="width:100%" >}}
```
Comment (alai97) [inline content/en/continuous_integration/guides/code_coverage.md:101]:
```suggestion
{{< img src="/continuous_integration/code_coverage_monitor.png" text="Test sessions code coverage tab" style="width:100%" >}}

For more information, see [CI Monitors][5].
```

File: content/en/continuous_integration/guides/code_coverage.md [hunk 16]
```diff
@@ -0,0 +1,111 @@
...
+
+## Further reading
+
+{{< partial name="whats-next/whats-next.html" >}}
+
+
+[1]: /continuous_integration/tests/
+[2]: /dashboards
+[3]: /notebooks
+[4]: /monitors
```
Comment (alai97) [inline content/en/continuous_integration/guides/code_coverage.md:111]:
```suggestion
[4]: /monitors
[5]: /monitors/types/ci
```

**Review by alai97 — COMMENTED** [review]
Looks good, some suggestions!

File: content/en/continuous_integration/guides/code_coverage.md [hunk 17]
```diff
@@ -0,0 +1,111 @@
...
+Jest includes istanbul by default, so you can simply do:
+
+```json
+{
+  "scripts": {
+    "test:coverage": "jest --coverage"
+  }
+}
+```
+
```
Comment (juan-fernandez) [inline content/en/continuous_integration/guides/code_coverage.md:58]:
```suggestion

Then, use:

```

File: content/en/continuous_integration/guides/code_coverage.md (hunk 4 above)
Comment (juan-fernandez) [reply content/en/continuous_integration/guides/code_coverage.md:59]: yeah I guess I can use the same `Then, use`

**Comment by juan-fernandez** [comment]
thanks @alai97 ! I applied all your feedback

**Comment by juan-fernandez** [comment]
ℹ️ I'll be adding an extra tab for JUnit XML uploads, which should also support code coverage

**Review by alai97 — APPROVED** [review]
Looks good, thanks for adding another tab!

**Review by alai97 — APPROVED** [review]

**Comment by juan-fernandez** [comment]
this is good to be merged now, right @alai97 ? 😄

**Review by rtrieu — APPROVED** [review]
Looks good!

**Pull request merged**